logstash-core/src/main/java/org/logstash/config/ir/compiler/Dataset.java

```diff
@@ -0,0 +1,438 @@
+package org.logstash.config.ir.compiler;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import org.jruby.RubyArray;
+import org.jruby.RubyHash;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+/**
+ * A data structure backed by a {@link RubyArray} that is lazily filled with
```

Comment:
This javadoc doesn't inform me as to the purpose of this class. It helps me understand an implementation detail (lazy, backed by RubyArray), but doesn't tell me what Dataset is used for.